src/test/java/com/google/api/generator/engine/ast/AssignmentExprTest.java

```diff
@@ -101,6 +101,42 @@ public void writeAssignmentExpr_nullObjectValuePrimitiveType() {
     assertInvalidAssignmentExpr(variableExpr, valueExpr);
   }
 
+  @Test
+  public void writeAssignmentExpr_primitiveToBoxedType() {
+    // [Constructing] `Boolean x = (boolean) true`
+    Variable variable = Variable.builder().setName("x").setType(TypeNode.BOOLEAN_OBJECT).build();
+    VariableExpr variableExpr =
+        VariableExpr.builder().setVariable(variable).setIsDecl(true).build();
+
+    Value value = PrimitiveValue.builder().setType(TypeNode.BOOLEAN).setValue("true").build();
+    Expr valueExpr = ValueExpr.builder().setValue(value).build();
+    assertValidAssignmentExpr(variableExpr, valueExpr);
+  }
+
+  @Test
+  public void writeAssignmentExpr_boxedToPrimitiveType() {
+    // [Constructing] `double x = (Double) y`
+    Variable lvariable = Variable.builder().setName("x").setType(TypeNode.DOUBLE).build();
```

Comment:
`lVariable`, same for `rVariable` below.

Reply:
Thanks for catching this, done.